core/modules/rproc/InfileMerger.cc

```diff
@@ -251,6 +267,56 @@ bool InfileMerger::_applySqlLocal(std::string const& sql) {
     return true;
 }
 
+
+size_t InfileMerger::_getResultTableSizeMB() {
+    std::string resultDb = _config.mySqlConfig.dbName;
+    std::string tableSizeSql = std::string("SELECT table_name, ")
+                             + "round(((data_length + index_length) / 1048576), 2) as 'MB' "
+                             + "FROM information_schema.TABLES "
+                             + "WHERE table_schema = '" + _config.mySqlConfig.dbName
+                             + "' AND table_name = '" + _mergeTable + "'";
+    LOGS(_log, LOG_LVL_DEBUG, "Checking ResultTableSize " << tableSizeSql);
+    std::lock_guard<std::mutex> m(_sqlMutex);
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    if (not _sqlConn.get()) {
+        _sqlConn = std::make_shared<sql::SqlConnection>(_config.mySqlConfig, true);
+        if (not _sqlConn->connectToDb(errObj)) {
+            _error = util::Error(errObj.errNo(),
+                     _queryIdStr + "Error connecting to db: " + errObj.printErrMsg(),
+                     util::ErrorCode::MYSQLCONNECT);
+            _sqlConn.reset();
+            LOGS(_log, LOG_LVL_ERROR, "InfileMerger error: " << _error.getMsg());
+            return 0;
+        }
+        LOGS(_log, LOG_LVL_DEBUG, _queryIdStr <<"InfileMerger " << (void*) this << " connected to db");
+    }
+    if (not _sqlConn->runQuery(tableSizeSql, results, errObj)) {
+        _error = util::Error(errObj.errNo(), "error getting size sql: " + errObj.printErrMsg(),
+                       util::ErrorCode::MYSQLEXEC);
+        LOGS(_log, LOG_LVL_ERROR, _queryIdStr << "result table size error: " << _error.getMsg());
+        return false;
+    }
+
+    // There should only be 1 row
+    auto iter = results.begin();
+    if (iter == results.end()) {
+        LOGS(_log, LOG_LVL_ERROR, _queryIdStr << "result table size no rows returned " << _mergeTable);
+        return 0;
+    }
+    auto& row = *iter;
+    std::string tbName = row[0].first;
+    std::string tbSize = row[1].first;
+    size_t sz;
+    std::stringstream ss;
+    ss << tbSize;
+    ss >> sz;
```

Comment:
This is sort of inefficient (streams are very heavy), I'd try `std::stoul()` instead.